File: Lab3/task2/partB/Supplier.cs
public class Supplier
{
    public int SupplierId { get; set; }
    public string? CompanyName { get; set; }
    public string? Street { get; set; }
    public string? City { get; set; }

    public int ProductId { get; set; }
    public Product? Product { get; set; }
}

File: Lab4/task2/HibernateJPALab/src/main/java/org/example/Supplier.java
package org.example;

import jakarta.persistence.*;

import java.util.ArrayList;
import java.util.List;

@Entity
public class Supplier extends Company {
    private String bankAccountNumber;

    @OneToMany(mappedBy = "supplier", cascade = CascadeType.ALL)
    private List<Product> products = new ArrayList<>();

    public Supplier() {}

    public Supplier(String companyName, String street, String city, String zipCode, String bankAccountNumber) {
        super(companyName, street, city, zipCode);
        this.bankAccountNumber = bankAccountNumber;
    }

    public String getBankAccountNumber() {
        return bankAccountNumber;
    }

    public void addProduct(Product product) {
        products.add(product);
        product.setSupplier(this);
    }

    public List<Product> getProducts() {
        return products;
    }
}
